transactor/transact.go

```diff
@@ -320,6 +321,18 @@ func (tx *Transaction) complete() {
 			logrus.Debugf("transactor(%d): abort succeeded", tx.ID)
 		}
 	}
+
+	// Update transaction record.
+	_, err = dal.SetTransaction(tx.Engine, &dal.Transaction{
+		ID:        tx.ID,
+		StartTime: tx.StartTime,
+		EndTime:   tx.EndTime,
+		Error:     tx.Error,
```

Comment:
Question. If, say, tx.Error is nil, but we pick up err!=nil during the commit, then err doesn't need to go into the Transaction record? Just wanted to check.